comp/metadata/packagesigning/utils/packagesignigutils.go

```diff
@@ -0,0 +1,107 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package utils defines shared methods in package signing component
+package utils
+
+import (
+	"os"
+	"runtime"
+)
+
+// GetLinuxPackageSigningPolicy returns the global GPG signing policy of the host
+func GetLinuxPackageSigningPolicy() (bool, bool) {
+	if runtime.GOOS == "linux" {
+		pkgManager := GetPackageManager()
+		switch pkgManager {
+		case "apt":
+			return getNoDebsig(), false
+		case "yum", "dnf", "zypper":
+			return getMainGPGCheck(pkgManager)
+		default: // should not happen, tested above
+			return false, false
+		}
+	}
+	return false, false
+}
+
+const (
+	aptPath  = "/etc/apt"
+	yumPath  = "/etc/yum"
+	dnfPath  = "/etc/dnf"
+	zyppPath = "/etc/zypp"
+)
+
+// GetPackageManager is a lazy implementation to detect if we use APT or YUM (RH or SUSE)
+func GetPackageManager() string {
+	if _, err := os.Stat(aptPath); err == nil {
+		return "apt"
+	} else if _, err := os.Stat(yumPath); err == nil {
+		return "yum"
+	} else if _, err := os.Stat(dnfPath); err == nil {
+		return "dnf"
+	} else if _, err := os.Stat(zyppPath); err == nil {
+		return "zypper"
+	}
+	return ""
+}
+
+// CompareRepoPerKeys is a method used on tests
+func CompareRepoPerKeys(a, b map[string][]Repositories) []string {
```

Comment:
I might be wrong, but it looks like this method is only used in one test file. Should we move to that test file and not make it part of the public API for the utils package?

Reply:
I need it on both `utils` and `packagesigningimpl` packages, so I don't know if I can improve this (without adding more complexity to the code organisation)

Reply:
That is a very fair point. I guess one way would be to add to the new package `test_utils.go` and import that package for `utils` and `packagesigningimpl`. It is mainly for code hygiene since we do not want to expose the test function as part of the `utils`. I also understand that those changes could be a bit more painful. So I leave up to you 😄